examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/MainApplication.kt

```diff
@@ -44,6 +47,25 @@ class MainApplication : Application(), UpdatedCustomerInfoListener {
         ).show()
         Log.d("CustomerInfoListener", "$message: $customerInfo")
     }
+
+    override fun handleRCBillingPurchaseRedemption(
+        redemptionStarter: RedeemRCBillingPurchaseListener.RedemptionStarter,
+    ) {
```

Comment:
> Their Activity's onCreate()/onNewIntent() I guess, depending on the launchMode?

Well, processing the deep link will be initiated from there, that's correct. However, to process the redemption link they can implement the `RedeemRCBillingPurchaseListener` interface any place they want. They just need to set the listener in the `Purchases.sharedInstance.redeemRCBillingPurchaseListener` at some point with that object. The current implementation caches the redemption token in memory until this listener is set.

> Is there any concern of a memory leak,

Well, depending on how long they hold the `redemptionStarter` there could be... This basically will hold some references to some internal classes of `Purchases`, but since most of those are singletons, it's mostly ok unless you reconfigure the SDK and keep an old reference to the `redemptionStarter`. I don't think we will be holding anything that would cause any big leaks on our side.

> What happens if the app never calls startRedemption()?

Nothing! We currently are not caching the links on disk so it won't survive restarts. We also won't try again from our side.